File: Python/tower_of_hanoi.py
# A simple python program to solve Tower of Hanoi problem
# By Anish Shilpakar
# Here I am using recursion to solve the problem

def TowerOfHanoi(n, source, destination, aux):
    if n == 1:
        print("Disk 1 : ",source, "-->", destination)
        return
    TowerOfHanoi(n-1, source, aux, destination)
    print("Disk",n,": ",source, "-->", destination)
    TowerOfHanoi(n-1, aux, destination, source)

def main():
    print("--------------------------------------------------")
    n = int(input("Enter the number of disks: "))
    print("------------------------------------------------\nDisk Movements for Tower of Hanoi Shown Below\n------------------------------------------------\nA: Source tower\nB: Auxillary tower\nC: Destination tower\n------------------------------------------------\n")
    # A is the source disk, C is the destination and B is the auxiliary disk initially
    TowerOfHanoi(n, 'A', 'C', 'B')
    print("------------------------------------------------\nDONE :)\n------------------------------------------------")

if __name__ == "__main__":
    main()

File: TowerOfHanoi.cpp
//c++ program to implement tower of hanoi problem
#include<iostream>
using namespace std;

//recursive funtion
void TowerOfHanoi(int num,char A,char B,char C){
  if(num>0){
    TowerOfHanoi(num-1, A, C, B);
    cout<<"Move a disk "<<num<<" from "<<" "<<A<<" to"<<" "<<C<<endl;
    TowerOfHanoi( num-1, B, A, C);
  }
}

//main funtion
int main(){
  int numOfDisk;
  cout<<"Enter the no. of disks"<<endl;
  cin>>numOfDisk;
  
  //calling recursive funtion
  TowerOfHanoi(numOfDisk,'A','B','C');//A is the source rod , C is destination rod ,B is auxiliary rod
  cout<<endl;
}
